python/semantic_kernel/connectors/memory/astradb/astra_client.py

```diff
@@ -0,0 +1,159 @@
+            }
+        }}
+        result = asyncio.run(self._run_query(self.request_base_url, query))
+        return True if result["status"]["ok"] == 1 else False
+
+    def delete_collection(self, collection_name: str):
+        query = {"deleteCollection": {"name": collection_name}}
+        result = asyncio.run(self._run_query(self.request_base_url, query))
+        return True if result["status"]["ok"] == 1 else False
+
+    def _build_collection_query(self, collection_name: str):
+        return f"{self.request_base_url}/{collection_name}"
+
+    def find_documents(self,
+                       collection_name: str,
+                       filter: Optional[Dict] = None,
+                       vector: Optional[List[float]] = None,
+                       limit: Optional[int] = None,
+                       include_vector: Optional[bool] = None,
+                       include_similarity: Optional[bool] = None) -> List[Dict]:
+        find_query = {}
+
+        if filter is not None:
+            find_query["filter"] = filter
+
+        if vector is not None:
+            find_query["sort"] = {
+                "$vector": vector
+            }
+
+        if include_vector is not None and include_vector == False:
+            find_query["projection"] = {
+                "$vector": 0
+            }
+
+        if limit is not None:
+            find_query["options"] = {
+                "limit": limit
+            }
+
+        if include_similarity is not None:
+            if "options" in find_query:
+                find_query["options"]["includeSimilarity"] = int(
+                    include_similarity)
+            else:
+                find_query["options"] = {
+                    "includeSimilarity": int(include_similarity)
+                }
+
+        query = {"find": find_query}
+        result = asyncio.run(self._run_query(
+            self._build_collection_query(collection_name), query))
+        return result["data"]["documents"]
+
+    def insert_document(self, collection_name: str, document: Dict) -> str:
+        query = {"insertOne": {"document": document}}
+        result = asyncio.run(self._run_query(
+            self._build_collection_query(collection_name), query))
+        return result["status"]["insertedIds"][0]
+
+    def insert_documents(self, collection_name: str, documents: List[Dict]) -> List[str]:
+        query = {"insertMany": {"documents": documents}}
+        result = asyncio.run(self._run_query(
+            self._build_collection_query(collection_name), query))
+        return result["status"]["insertedIds"]
+
+    def update_document(self, collection_name: str, filter: Dict, update: Dict, upsert: bool = True) -> Dict:
+        query = {"findOneAndUpdate": {
+            "filter": filter,
+            "update": update,
+            "options": {
+                "returnDocument": "after",
+                "upsert": upsert
+            }
+        }}
+        result = asyncio.run(self._run_query(
```

Comment:
do not use asyncio.run inside here, just make the method async and do `await self._run_query`

Reply:
Done